build/parsePrep.c

```diff
@@ -345,7 +349,7 @@ static int doSetupMacro(rpmSpec spec, const char *line)
     }
 
     if (!createDir) {
-	rasprintf(&buf, "cd '%s'", spec->buildSubdir);
+	buf = rpmExpand("%{?buildsubdir:cd '%{buildsubdir}'}", NULL);
```

Comment:
We didn't need all these conditionals before, why do we need them now?

Reply:
I mean, this is inside %setup macro, %buildsubdir is always defined there. The special case of build-in-place jumps over this whole section.

Reply:
As an aside, the build-in-place logic seems a bit broken to begin with: it pushes macro will NULL body, which isn't the right thing, it shouldn't define buildsubdir at all then (but that's not the fault of this patchset of course)